From ba3251403f26d0f150d233dbeb423f868cc1fb4a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Anders=20F=20Bj=C3=B6rklund?= <anders.f.bjorklund@gmail.com>
Date: Tue, 13 Mar 2018 12:25:57 +0100
Subject: [PATCH] Cope with mem RRD missing cached/slab/buffers

When introducing the new values for "slab",
there is period when old values are missing...

Make sure to count those as zero properly,
so that the total amount of memory lines up
---
 graph.d/mem_report.php | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/graph.d/mem_report.php b/graph.d/mem_report.php
index 28c64e45..fc18d1c0 100644
--- a/graph.d/mem_report.php
+++ b/graph.d/mem_report.php
@@ -50,19 +50,19 @@ function graph_mem_report ( &$rrdtool_graph ) {
     $bmem_used_cdef = "CDEF:'bmem_used'='bmem_total','bmem_free',-,'bmem_cached',-";
 
     if (file_exists("$rrd_dir/mem_shared.rrd")) {
-       $bmem_used_cdef .= ",'bmem_shared',-";
+       $bmem_used_cdef .= ",'bmem_shared',UN,0,'bmem_shared',IF,-";
        $bmem_shared_defs = "DEF:'mem_shared'='${rrd_dir}/mem_shared.rrd':'sum':AVERAGE "
            ."CDEF:'bmem_shared'=mem_shared,1024,* ";
     }
 
     if (file_exists("$rrd_dir/mem_slab.rrd")) {
-       $bmem_used_cdef .= ",'bmem_slab',-";
+       $bmem_used_cdef .= ",'bmem_slab',UN,0,'bmem_slab',IF,-";
        $bmem_slab_defs = "DEF:'mem_slab'='${rrd_dir}/mem_slab.rrd':'sum':AVERAGE "
            ."CDEF:'bmem_slab'=mem_slab,1024,* ";
     }
 
     if (file_exists("$rrd_dir/mem_buffers.rrd")) {
-       $bmem_used_cdef .= ",'bmem_buffers',-";
+       $bmem_used_cdef .= ",'bmem_buffers',UN,0,'bmem_buffers',IF,-";
        $bmem_buffers_defs = "DEF:'mem_buffers'='${rrd_dir}/mem_buffers.rrd':'sum':AVERAGE "
            ."CDEF:'bmem_buffers'=mem_buffers,1024,* ";
     }
